src/Common/src/Interop/OSX/Interop.CoreFoundation.cs

```diff
@@ -52,10 +54,10 @@ private enum CFStringBuiltInEncodings : uint
         /// Creates a CFStringRef from a 8-bit String object. Follows the "Create Rule" where if you create it, you delete it.
         /// </summary>
         /// <param name="str">The string to get a CFStringRef for</param>
-        /// <returns>Returns a pointer to a CFString on success; otherwise, returns IntPtr.Zero</returns>
-        internal static CFStringRef CFStringCreateWithCString(string str)
+        /// <returns>Returns a valid SafeCreateHandle to a CFString on success; otherwise, returns an invalid SafeCreateHandle</returns>
+        internal static SafeCreateHandle CFStringCreateWithCString(string str)
         {
-            return CFStringCreateWithCString(IntPtr.Zero, str, CFStringBuiltInEncodings.kCFStringEncodingUTF8);
+            return new SafeCreateHandle(CFStringCreateWithCString(IntPtr.Zero, str, CFStringBuiltInEncodings.kCFStringEncodingUTF8));
```

Comment:
Yes, as long as they're IntPtrs, it should work fine, and that would be the preferred way.  There are a few places we've haven't done that, but they're the minority case (e.g. when the thing we're wrapping is actually a file descriptor and is still 32-bits on a 64-bit architecture).